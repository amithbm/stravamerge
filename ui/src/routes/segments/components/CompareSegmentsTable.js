import React from 'react'
import { Table } from 'react-bootstrap'

const CompareSegmentsTable = ({tableData}) => <Table striped>  
    <thead>
        <tr>
            <th>Segment</th>
            <th>Ride Time (seconds)</th>
            <th>Best Time (seconds)</th>
        </tr>
    </thead>
    <tbody>
    {tableData.map((item,index) =>
        <tr key={index}>
            <td>{item.name}</td> 
            <td>{item.actual}</td>
            <td>{item.best}</td>
        </tr>
    )}
    </tbody>
    <tfoot>
        <tr>
            <th>Total</th>
            <th>{tableData.map(i => i.actual).reduce((a,b) => a+b)}</th>
            <th>{tableData.map(i => i.best).reduce((a,b) => a+b)}</th>
        </tr>
    </tfoot>
</Table>

CompareSegmentsTable.propTypes = {
    tableData: React.PropTypes.array.isRequired
}

export default CompareSegmentsTable
