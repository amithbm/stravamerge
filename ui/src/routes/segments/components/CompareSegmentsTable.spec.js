import React from 'react'
import {shallow} from 'enzyme'
import CompareSegmentsTable from './CompareSegmentsTable'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<CompareSegmentsTable />', () => {
        it('renders as expected', () => {
            const tableData =[
                {
                    name: 'Some Segment',
                    actualTime: 250,
                    bestTime: 220
                },
                {
                    name: 'Another segment',
                    actualTime: 300,
                    bestTime: 200
                }
            ]
            const component =  shallow(<CompareSegmentsTable tableData={tableData} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

