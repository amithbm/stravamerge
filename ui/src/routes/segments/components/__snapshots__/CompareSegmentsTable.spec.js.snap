exports[`components <CompareSegmentsTable /> renders as expected 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={false}
  striped={true}>
  <thead>
    <tr>
      <th>
        Segment
      </th>
      <th>
        Ride Time (seconds)
      </th>
      <th>
        Best Time (seconds)
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Some Segment
      </td>
      <td />
      <td />
    </tr>
    <tr>
      <td>
        Another segment
      </td>
      <td />
      <td />
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th>
        Total
      </th>
      <th />
      <th />
    </tr>
  </tfoot>
</Table>
`;
